From 6cee9dd4b39082ec7d5afcb1f4c0b91832e294d9 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 7 Jun 2023 16:23:45 +0200
Subject: [PATCH] man: document that journalctl --user requires
 Storage=persistent

Fixes: #25061
(cherry picked from commit cad8fa471b9133f38fbd6ddb6ee143f361525e5d)

Related: RHEL-31070
---
 man/journalctl.xml    | 6 +++++-
 man/journald.conf.xml | 3 +++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/man/journalctl.xml b/man/journalctl.xml
index 29d06aaef8..6be628caf8 100644
--- a/man/journalctl.xml
+++ b/man/journalctl.xml
@@ -111,7 +111,11 @@
         </para>
 
         <para>The <option>--user</option> option affects how <option>--unit=</option> arguments are
-        treated. See <option>--unit=</option>.</para></listitem>
+        treated. See <option>--unit=</option>.</para>
+
+        <para>Note that <option>--user</option> only works if persistent logging is enabled, via the
+        <varname>Storage=</varname> setting in
+        <citerefentry><refentrytitle>journald.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>.</para></listitem>
       </varlistentry>
 
       <varlistentry>
diff --git a/man/journald.conf.xml b/man/journald.conf.xml
index 24cee4c8b2..a7c3da6653 100644
--- a/man/journald.conf.xml
+++ b/man/journald.conf.xml
@@ -95,6 +95,9 @@
         <filename>/var/log/journal/</filename>, as the <filename>systemd-journald@.service</filename> service
         file by default carries <varname>LogsDirectory=</varname>. To turn that off, add a unit file drop-in
         file that sets <varname>LogsDirectory=</varname> to an empty string.</para>
+
+        <para>Note that per-user journal files are not supported unless persistent storage is enabled, thus
+        making <command>journalctl --user</command> unavailable.</para>
         </listitem>
       </varlistentry>
 
